compiler/rustc_target/src/spec/base/apple/tests.rs

```diff
@@ -18,7 +18,7 @@ fn simulator_targets_set_abi() {
     ];
 
     for target in &all_sim_targets {
-        assert_eq!(target.abi, "sim")
+        assert_eq!(target.env, "sim")
```

Comment:
Could probably check both `.abi` and `.env` here to validate backwards compact doesn't regress mistakenly?